ls-main.go

```diff
@@ -135,10 +135,13 @@ func mainList(ctx *cli.Context) {
 		clnt, err := newClient(targetURL)
 		fatalIf(err.Trace(targetURL), "Unable to initialize target ‘"+targetURL+"’.")
 
-		st, err := clnt.Stat()
-		fatalIf(err.Trace(targetURL), "Unable to stat target ‘"+targetURL+"’.")
-
-		if st.Type.IsDir() {
+		var st *clientContent
+		if st, err = clnt.Stat(); err != nil {
+			// For aliases like ``mc ls s3`` it's acceptable to receive BucketNameEmpty error.
+			if _, bucketEmpty := err.Cause.(BucketNameEmpty); !bucketEmpty {
```

Comment:
You shouldn't access the probe errors internal semantics like this. It should be done as below to be consistent with other parts of the code so that the underlying API change or names doesn't affect the future functionalities. 

```
switch e := err.ToGoError().(type) {
case BucketNameEmpty:
```